### PR TITLE
Add \codepoint_category:n

### PR DESCRIPTION
**Pull request by josephwright**

### CODE REVIEW

**Comment by josephwright** [comment]
This is step one to addressing https://github.com/latex3/latex3/issues/1232 - step two is to provide word-by-word mappings.

**Comment by josephwright** [comment]
My main question is about the naming here: should it be `\codepoint_to_category:n`, for example, as there is no 'action' otherwise? Or `\codepoint_categorise:n`?

**Comment by muzimuzhi** [comment]
> My main question is about the naming here: should it be `\codepoint_to_category:n`, for example, as there is no 'action' otherwise? Or `\codepoint_categorise:n`?

I vote for `\codepoint_to_category:n`, partly because there're already other `\forma_to_formb:<specs>` functions, especially `\codepoint_to_nfd:n`.

**Comment by josephwright** [comment]
@muzimuzhi Otherwise looks OK?

File: l3kernel/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ this project uses date-based 'snapshot' version identifiers.
 
 ## [Unreleased]
 
+### Added
+- `\codepoint_to_category:n`
```
Comment (muzimuzhi) [inline l3kernel/CHANGELOG.md:11]: Styling: A newline is missing at the end, aka just before `## [2023-06-16]`.

**Review by PhelypeOleinik — APPROVED** [review]

File: l3kernel/l3unicode.dtx [hunk 2]
```diff
@@ -106,6 +106,21 @@
...
+%   \end{syntax}
+%   Expands to the Unicode general category identifier of the \meta{codepoint}.
+%   The general category identifier is a string made up of two letter
+%   characters, the first uppercase and the second lowercase. The uppercase
+%   letters divide codepoints into broader groups, which are then refined
+%   by the lowercase letter. For example, codepoints representing letters
+%   all have identifiers starting \texttt{L}, for example \texttt{Lu}
+%   (uppercase letter), \texttt{Lt} (titlecase letter), \emph{etc.}
+%   Full details are available in the documentation provided by the Unicode
+%   Consortium.
```
Comment (FrankMittelbach) [inline l3kernel/l3unicode.dtx:121]: if that has a canonical reference it would be nice to give it

**Review by FrankMittelbach — APPROVED** [review]

**Pull request merged**